addons/website_form/static/src/snippets/s_website_form/000.js

```diff
@@ -195,6 +195,8 @@ odoo.define('website_form.s_website_form', function (require) {
                         case 'redirect':
                             successPage = successPage.startsWith("/#") ? successPage.slice(1) : successPage;
                             if (successPage.charAt(0) === "#") {
+                                // Escape special characters to make the jQuery selector to work.
+                                successPage = '#' + $.escapeSelector(successPage.substring(1));
```

Comment:
Why is this not necessary for anchors in links? I don't see this code in `anchorSlide` widget?

Reply:
Actually in this case it's only the animation that doesnt work because of :
```js
        if (!utils.isValidAnchor(hash)) {
            return;
        }
```
in the `_onAnimateClick` function...
I agree that we also have to fix it there

Reply:
But if we do that we have to remove the `isValidAnchor` because we will allow invalid anchor..
Since some anchor name are generated by the anchor option with the first characters of a snippet content, I guess that we can remove it as everything is possible... What do you thing about it ?

Reply:
Here is what I found investigating:

1. Initially, we allowed everything as anchor and it was possible it crashed
2. We did this: https://github.com/odoo/odoo/commit/0abfaeda96c2eaa868cc7fc5fa1926dfa90fc420 to ensure to only use valid characters
3. Then did: https://github.com/odoo/odoo/commit/b492bde6a121be1c15ed90ce0827fcfd72a12f5c to create an util for what was done (`isValidAnchor` etc)
4. But then later we did this: https://github.com/odoo/odoo/commit/f615cfda89dae2e9a550d25868b11bc06b97cfcb allowing all characters again but they are encoded once they are copied... -> I guess here the job was not fully done and should have removed the `isValidAnchor` calls.

What I would do is:
1. Ensure what I say here is correct
2. In the stable version where (4) was done, complete the job:
    2.1 Remove the uses of `isValidAnchor` (but keep the definition in stable)
    2.2 Encode the name the same way it is normally done (see `_text2Anchor` in (4)?) if necessary? Should avoid the use of `$.escapeSelector` I guess. I remember that when we did (4), redirecting to anchor containing emoji worked without the need of this `$.escapeSelector`

Reply:
What I understand is that `encodeURIComponent` (used in `_text2Anchor`) is useful when the anchor is in an other page 
to keep the url valid, and so we dont need it in `_onAnimateClick` because it's used only to scroll in the same page.
However, because of the jQuery selector used in `this._scrollTo($anchor, scrollValue);` we need to use `$.escapeSelector` in this case.
So I would add  `hash = '#' + $.escapeSelector(hash.substring(1));` in `_onAnimateClick`, so the jQuery selector will be valid.
 And remove `isValidAnchor` because it will not let pass the escaped hash and as I explained we dont need it to scroll in the same page. I tested it and it will continue to allow smiley 👻 in anchor names but also anchor name as "20%-Discount"

Reply:
Seems good to me :+1: I'll let @rdeodoo review or wait for me in 2 weeks :+1:

Reply:
note that because I removed `isValidAnchor`, I had to add `if (!hash.length) { return ; };` to avoid a crash if the href is `href="#"`,

Reply:
@rdeodoo The PR is ready to be reviewed

Reply:
> It is sad to see that QSM [did a complete investigation](https://github.com/odoo/odoo/pull/82059#discussion_r776972788) about the history of this line but you did not add it in the commit message..
> 
> I actually did the same job again myself, finding [b492bde](https://github.com/odoo/odoo/commit/b492bde6a121be1c15ed90ce0827fcfd72a12f5c) and [0abfaed](https://github.com/odoo/odoo/commit/0abfaeda96c2eaa868cc7fc5fa1926dfa90fc420) while it could have been written inside the commit message..

I thought that the conversion in this PR would be enough to avoid having to do a whole re-investigation on the issue.